OSSM-5726 Distributed Tracing Docs Section Updates and Tempo content (2.5 Release) (#70517)
openshift/openshift-docs: modules/ossm-configuring-distr-tracing-tempo.adoc

////
This module is included in the following assemblies:
* service_mesh/v2x/ossm-observability.adoc
////

:_mod-docs-content-type: PROCEDURE
[id="ossm-configuring-distr-tracing-tempo_{context}"]
= Configuring the distributed tracing platform (Tempo)

You can expose tracing data to the {TempoName} stack by appending a named element and `zipkin` provider to the `spec.meshConfig.extensionProviders` specification. Then, a telemetry custom resource configures Istio proxies to collect trace spans and send them to the Tempo distributor service endpoint.

> The following example configures the `ServiceMeshControlPlane` resource to define an extensionProvider `zipkin` for sending tracing spans to a TempoStack. You can create a TempoStack instance in the `tracing-system` namespace _after_ creating the `ServiceMeshControlPlane` and the `ServiceMeshMemberRoll` resources.

Reviewer: ```suggestion
You can create a TempoStack instance in the `tracing-system` namespace _after_ creating the `ServiceMeshControlPlane` and the `ServiceMeshMemberRoll` resources.
```

Reviewer: I'm wondering if you add "as shown in the following example", maybe you can avoid repeating some of the same procedural concepts in this 2nd paragraph?

.Prerequisites

You have installed the {TempoOperator} and {SMProductName} Operator in the `openshift-operators` namespace and have created namespaces such as `istio-system` and `tracing-system`.

.Procedure

. Configure the `ServiceMeshControlPlane` resource to define an extension provider:
+
[source,yaml]
----
kind: ServiceMeshControlPlane
apiVersion: maistra.io/v2
metadata:
  name: basic
  namespace: istio-system
spec:

> ...

Reviewer: ```suggestion
# ...
```

Reviewer: Nice job using the ellipses here. Per the [contributing](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#yaml-formatting) guidance, comment out ellipses when truncating YAML to avoid creating a document end marker in the YAML.

Reviewer: Is there a place to end the ellipses?

  meshConfig:
    extensionProviders:
      - name: tempo
        zipkin:
          service: tempo-sample-distributor.tracing-system.svc.cluster.local
          port: 9411
  tracing:
    sampling: 10000
    type: None <1>
  version: v2.5
---
kind: ServiceMeshMemberRoll
apiVersion: maistra.io/v1
metadata:
  name: default
  namespace: istio-system
spec:
  members:
    - tracing-system
----

> <1> The `spec.tracing.type` setting defines a Distributed Tracing deprecated Jaeger instance. Set `spec.tracing.type` to `None` when connecting to a TempoStack using an `extensionProvider` setting.

Reviewer: ```suggestion
<1> The `spec.tracing.type` setting defines a deprecated Distributed Tracing Jaeger instance. Set `spec.tracing.type` to `None` when connecting to a TempoStack using an `extensionProvider` setting.
```

Reviewer: I just think this flows better if you move deprecated around in the sentence. Wdyt?

+
[NOTE]
====
Create a TempoStack instance _after_ creating the `ServiceMeshControlPlane` and the `ServiceMeshMemberRoll` resources.
====

> +

Reviewer: ```suggestion

```

Reviewer: I dont think this "+" does anything for you. You only needed it before the NOTE to keep it in alignment with the preceding step.

. Configure the Kiali resource specification to enable a Kiali workload traces dashboard. You can use the dashboard to view tracing query results.
+
[source,yaml]
----
apiVersion: kiali.io/v1alpha1
kind: Kiali

> ...

Reviewer: Same comment here about the # with the ellipses.

spec:  
  external_services:
    tracing:
      enabled: true
      in_cluster_url: 'http://tempo-sample-query-frontend.tracing-system.svc.cluster.local:16686'
      url: '[Tempo query frontend Route url]'
----

> +

Reviewer: ```suggestion

```

Reviewer: Same comment as above. This isn't needed you can just carry on with the next step.

. Create a TempoStack instance using the Red Hat {TempoOperator} in the `tracing-system` namespace. For more information, see "Installing the distributed tracing platform (Tempo)" in the {product-title} documentation.

> +

Reviewer: ```suggestion

```

Reviewer: Same as other comments.

. Apply a Telemetry custom resource for {SMProductShortName} to start the Tempo provider setting.
+
[source,yaml]
----
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: mesh-default
  namespace: istio-system
spec:
  tracing:
  - providers:
    - name: tempo

>     randomSamplingPercentage: 100

Reviewer: Just checking, is the alignment of `randomSamplingPercentage` OK? I was thinking maybe the "r" needs to align with the "n" (this is not because I am familiar with this resource, just weird to see the parameter aligned with the "-"

----

> You can also create Istio gateway and virtual service resources to expose an OpenShift route for accessing the Tempo Jaeger Query console.

Reviewer: ```suggestion
You can also create an Istio gateway and virtual service resources to expose an {product-title} route for accessing the Tempo Jaeger Query console.
```

Reviewer: Also wondering, this seems sort of like an optional suggestion-type statement. Maybe its best-served in a NOTE? wdyt?

Reviewer: And also wondering if there's an additional resource link for doing this type of thing?

Author: That is a post release task.

You can create a TempoStack instance in the `tracing-system` namespace _after_ creating the `ServiceMeshControlPlane` and the `ServiceMeshMemberRoll` resources.

.Prerequisites

* You have installed the {TempoOperator} and {SMProductName} Operator in the `openshift-operators` namespace.
* You have created namespaces such as `istio-system` and `tracing-system`.

.Procedure

. Configure the `ServiceMeshControlPlane` resource to define an extension provider:
+
[source,yaml]
----
kind: ServiceMeshControlPlane
apiVersion: maistra.io/v2
metadata:
  name: basic
  namespace: istio-system
spec:
# ...
  meshConfig:
    extensionProviders:
      - name: tempo
        zipkin:
          service: tempo-sample-distributor.tracing-system.svc.cluster.local
          port: 9411
  tracing:
    sampling: 10000
    type: None <1>
  version: v2.5
---
kind: ServiceMeshMemberRoll
apiVersion: maistra.io/v1
metadata:
  name: default
  namespace: istio-system
spec:
  members:
    - tracing-system
----

> <1> The `spec.tracing.type` setting defines a deprecated Distributed Tracing Jaeger instance. Set `spec.tracing.type` to `None` when connecting to a TempoStack using an `extensionProvider` setting.

Reviewer 2: ```suggestion
<1> The `spec.tracing.type` setting defines a deprecated distributed tracing Jaeger instance. Set `spec.tracing.type` to `None` when connecting to a TempoStack using an `extensionProvider` setting.
```
Not going to hold up merge, but the caps here should be made the same throughout at some point.
